third_party/2and3/flask/app.pyi

```diff
@@ -0,0 +1,139 @@
+# Stubs for flask.app (Python 3.6)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from .blueprints import Blueprint
+from .config import Config, ConfigAttribute
+from .ctx import AppContext, RequestContext, _AppCtxGlobals
+from .globals import _request_ctx_stack, g, request, session
+from .helpers import _PackageBoundObject, find_package, get_debug_flag, get_env, get_flashed_messages, get_load_dotenv, locked_cached_property, url_for
+from .logging import create_logger
+from .sessions import SecureCookieSessionInterface
+from .signals import appcontext_tearing_down, got_request_exception, request_finished, request_started, request_tearing_down
+from .templating import DispatchingJinjaLoader, Environment
+from .wrappers import Request, Response
+from .testing import FlaskClient
+from typing import Any, Callable, ContextManager, Dict, List, Optional, Type, TypeVar, Union
+
+def setupmethod(f: Any): ...
+
+_T = TypeVar('_T')
+
+class Flask(_PackageBoundObject):
+    request_class: type = ...
+    response_class: type = ...
+    jinja_environment: type = ...
+    app_ctx_globals_class: type = ...
+    config_class: type = ...
+    testing: Any = ...
+    secret_key: Any = ...
+    session_cookie_name: Any = ...
+    permanent_session_lifetime: Any = ...
+    send_file_max_age_default: Any = ...
+    use_x_sendfile: Any = ...
+    json_encoder: Any = ...
+    json_decoder: Any = ...
+    jinja_options: Any = ...
+    default_config: Any = ...
+    url_rule_class: type = ...
+    test_client_class: type = ...
+    test_cli_runner_class: type = ...
+    session_interface: Any = ...
+    import_name: str = ...
+    template_folder: str = ...
+    root_path: Any = ...
+    static_url_path: Any = ...
+    static_folder: str = ...
+    instance_path: Any = ...
+    config: Dict[str, Any] = ...
+    view_functions: Any = ...
+    error_handler_spec: Any = ...
+    url_build_error_handlers: Any = ...
+    before_request_funcs: Dict[Optional[str], List[Callable[[], Any]]] = ...
+    before_first_request_funcs: List[Callable[[], None]] = ...
+    after_request_funcs: Dict[Optional[str], List[Callable[[Response], Response]]] = ...
+    teardown_request_funcs: Dict[Optional[str], List[Callable[[Optional[Exception]], Any]]] = ...
+    teardown_appcontext_funcs: List[Callable[[Optional[Exception]], Any]] = ...
+    url_value_preprocessors: Any = ...
+    url_default_functions: Any = ...
+    template_context_processors: Any = ...
+    shell_context_processors: Any = ...
+    blueprints: Any = ...
+    extensions: Any = ...
+    url_map: Any = ...
+    subdomain_matching: Any = ...
+    cli: Any = ...
```

Comment:
`_get_current_object` is private, but it is often accessed on `Flask`, so I thin kit makes sense to add it.

Reply:
Actually it's called on LocalProxy that is wrapping Flask. I've created a FlaskLocalProxy class that will allow `_get_current_object` on `current_app` however it will forbid `current_app._get_current_object()._get_current_object()`